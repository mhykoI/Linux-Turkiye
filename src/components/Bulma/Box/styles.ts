import { Styles } from 'react-jss';
import globalStyles from '../../../styles/global';

const styles: Styles = {
  boxCustom: {
    // ...globalStyles.glass,
  },
};

export default styles;
