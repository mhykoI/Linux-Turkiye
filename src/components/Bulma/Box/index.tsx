import React from 'react';
import clsx from 'clsx';
import { createUseStyles } from 'react-jss';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface BoxProps {
  content?: boolean;
}

const Box: React.FC<BoxProps> = ({ content, children }) => {
  const classes = useStyles();

  return (
    <div className={clsx('box', classes.boxCustom, { ['content']: !!content })}>
      {children}
    </div>
  );
};

export default Box;
