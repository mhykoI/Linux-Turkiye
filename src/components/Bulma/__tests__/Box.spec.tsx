import React from 'react';
import renderer from 'react-test-renderer';
import Box from '../Box';

describe('<Box />', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<Box content>This is a Box!</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
