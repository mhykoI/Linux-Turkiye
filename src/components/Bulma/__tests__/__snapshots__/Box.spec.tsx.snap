// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render correctly 1`] = `
<div
  className="box boxCustom-0-2-1 content"
>
  This is a Box!
</div>
`;
